src/main/java/model/http/SessionManager.java

```diff
@@ -0,0 +1,30 @@
+package model.http;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+
+public class SessionManager {
+    private static SessionManager sessionManager = new SessionManager();
+
+    private SessionManager() {}
+
+    public static SessionManager getInstance() {
+        return sessionManager;
+    }
+
+    private static final Map<String, CustomHttpSession> SESSIONS = new HashMap<>();
+
+    public void add(final CustomHttpSession session) {
+        SESSIONS.put(session.getId(), session);
+    }
+
+    public Optional<CustomHttpSession> findSession(final String id) {
+        return Optional.of(SESSIONS.getOrDefault(id, null));
+    }
```

Comment:
사소하지만 getOrDefault 대신 get을 사용해도 괜찮을 것 같아요 :)

```suggestion
    public Optional<CustomHttpSession> findSession(final String id) {
        return Optional.of(SESSIONS.get(id));
    }
```